src/simd_radix2_fft.h

```diff
@@ -44,22 +62,31 @@ namespace simd {
  * x <- x + r * y
  * y <- x - r * y
  *
- * @param rp1 coefficient `r` plus one
+ * @param ct_case coefficient case
  * @param c a register stores coefficient `r`
  * @param x working register
  * @param y working register
- * @param q modular
  */
 template <typename T>
-inline void butterfly_ct(T rp1, VecType c, VecType* x, VecType* y, T q)
+inline void
+butterfly_ct(CtGsCase ct_case, const VecType& c, VecType& x, VecType& y)
 {
-    VecType z = (rp1 == 2) ? *y : mod_mul(c, *y, q);
-    if (rp1 < q) {
-        *y = mod_sub(*x, z, q);
-        *x = mod_add(*x, z, q);
-    } else { // i.e. r == q - 1
-        *y = mod_add(*x, z, q);
-        *x = mod_sub(*x, z, q);
+    VecType z = y;
+    switch (ct_case) {
+    case CtGsCase::SIMPLE:
+        y = mod_sub<T>(x, z);
+        x = mod_add<T>(x, z);
+        break;
+    case CtGsCase::EXTREME:
+        y = mod_add<T>(x, z);
+        x = mod_sub<T>(x, z);
+        break;
+    case CtGsCase::NORMAL:
+    default:
```

Comment:
Remove the `default` case.

Since we are using an enum, the compiler can statically enforce that we cover all the enum values.
If one day we add a new case, we will have a compiler error if we don't update the switch, e.g.:

```
error: enumeration value 'NEW_CASE' not handled in switch [-Werror=switch-enum]
```

Reply:
Thanks. I will update it